classes/ObjectModel.php

```diff
@@ -1927,7 +1927,7 @@ public function hydrate(array $data, $id_lang = null)
     {
         $this->id_lang = $id_lang;
         if (isset($data[$this->def['primary']])) {
-            $this->id = $data[$this->def['primary']];
+            $this->id = (int) $data[$this->def['primary']];
```

Comment:
It was this line the responsible ? Woha 😄  nice catch

Reply:
Yes !